pkgs/development/python-modules/pycdio/default.nix

```
@@ -29,23 +32,32 @@ buildPythonPackage rec {
     patchShebangs .
   '';
 
+  build-system = [ setuptools ];
+
   nativeBuildInputs = [
-    nose
-    pkgs.pkg-config
-    pkgs.swig
+    pkg-config
+    swig
   ];
+
   buildInputs = [
-    setuptools
-    pkgs.libcdio
-  ] ++ lib.optional stdenv.isDarwin pkgs.libiconv;
-
-  # Run tests using nosetests but first need to install the binaries
-  # to the root source directory where they can be found.
-  checkPhase = ''
-    ./setup.py install_lib -d .
-    nosetests
+    libcdio
+    libiconv
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  preCheck = ''
+    substituteInPlace test/test-cdtext.py \
+      --replace-fail "assertEquals" "assertEqual"
   '';
 
+  pytestFlagsArray = [
+    "test/test-cdio.py"
+    "test/test-cdtext.py"
+    "test/test-iso.py"
+    "test/test-isocopy.py"
+  ];
```

Can’t this just be `"test"` (or `"test/"`)?

---

Probably, we can't. The upstream uses `-` instead of `_`.
https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html#conventions-for-python-test-discovery

---

This works and would be more robust to future additions of tests:

```suggestion
  pytestFlagsArray = [ "test/test-*.py" ];
```